src/IO/PeekableReadBuffer.h

```diff
@@ -38,6 +39,11 @@ class PeekableReadBuffer : public BufferWithOwnMemory<ReadBuffer>
             peeked_size = 0;
         }
         checkpoint = pos;
+
+        // FIXME: we are checking checkpoint existence in few places (rollbackToCheckpoint/dropCheckpoint)
+        // by simple if(checkpoint) but checkpoint can be nullptr after
+        // setCheckpoint called on empty (non initialized/eof) buffer
+        // and we can't just use simple if(checkpoint)
```

Comment:
It is the reason why Template format doesn't work with Kafka. 
I've fixed the issue by adding one more flag but after that I get to a very similar problem as segfault with empty message in RowBinaryWithNamesAndTypes

Reply:
If i fix that i get the following segfault
```
########################################
(version 20.5.1.1) (from thread 25) (no query) Received signal Segmentation fault (11).
Address: NULL pointer. Access: read. Address not mapped to object.
Stack trace: 0x15c8d9c1 0x15c8db3d 0x1dfbb955 0x1dfb2e05 0x1de7289f 0x1dd79807 0x1d8f80c4 0x1d8f6942 0x1c9c92fd 0x1c9ea01c 0x1c9e9fc7 0x1d8cc3fa 0x1d8cb509 0x1d8ce978 0x1d8ce93d 0x1d8ce8ed 0x1d8ce8bd 0x1d8cda1e 0x15c4f055 0x15c4dfb5 0x1cbcfe12 0x1cbd6f96 0x1cbd1873 0x1cbd4c58 0x1cbd4c2d 0x1cbd4bf1 0x1cbd4bc2 0x1cbd4abc
/src/IO/ReadHelpers.cpp:92: DB::checkString(char const*, DB::ReadBuffer&) @ 0x15c8d9c1 in /usr/bin/clickhouse
/src/IO/ReadHelpers.cpp:119: DB::assertString(char const*, DB::ReadBuffer&) @ 0x15c8db3d in /usr/bin/clickhouse
/src/IO/ReadHelpers.h:181: DB::assertString(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::ReadBuffer&) @ 0x1dfbb955 in /usr/bin/clickhouse
/src/Processors/Formats/Impl/TemplateRowInputFormat.cpp:171: DB::TemplateRowInputFormat::readRow(std::__1::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn> > >&, DB::RowReadExtension&) @ 0x1dfb2e05 in /usr/bin/clickhouse
```